apps/03TicTacToe.js

```diff
@@ -24,23 +24,81 @@ function printBoard() {
 }
 
 function horizontalWin() {
-  // Your code here
+
+//  Your code here
+  if ((board[0][0] === 'X' && board[0][1] === 'X' && board[0][2] === 'X' ) || (board[0][0] === 'O' && board[0][1] === 'O' && board[0][2] === 'O')) {
+      console.log(playerTurn + " !!! WINS!!!    ...Resetting Game");
+
+        board = [
+          [' ', ' ', ' '],
+          [' ', ' ', ' '],
+          [' ', ' ', ' ']
+        ];
+return true;
+  } else
+  return false;
 }
 
 function verticalWin() {
-  // Your code here
+        if  ((board[0][0] === 'X' && board[1][0] === 'X' && board[2][0] === 'X' )
+          || (board[0][1] === 'X' && board[1][1] === 'X' && board[2][1] === 'X')
+          || (board[0][2] === 'X' && board[1][2] === 'X' && board[2][2] === 'X')
+          || (board[0][0] === 'O' && board[1][0] === 'O' && board[2][0] === 'O')
+          || (board[0][1] === 'O' && board[1][1] === 'O' && board[2][1] === 'O')
+          || (board[0][2] === 'O' && board[1][2] === 'O' && board[2][2] === 'O'))  {
+      console.log(playerTurn + " !!! WINS!!!    ...Resetting Game.");
+
+        board = [
+          [' ', ' ', ' '],
+          [' ', ' ', ' '],
+          [' ', ' ', ' ']
+        ];
+  return true;
+  } else
+  return false;
 }
 
 function diagonalWin() {
   // Your code here
+  if  ((board[0][0] === 'X' && board[1][1] === 'X' && board[2][2] === 'X' )
+    || (board[0][2] === 'X' && board[1][1] === 'X' && board[2][0] === 'X')
+    || (board[0][0] === 'O' && board[1][1] === 'O' && board[2][2] === 'O')
+    || (board[0][2] === 'O' && board[1][1] === 'O' && board[2][0] === 'O')) {
+  console.log(playerTurn + " !!! WINS!!!    ...Resetting Game.");
+
+   board = [
+    [' ', ' ', ' '],
+    [' ', ' ', ' '],
+    [' ', ' ', ' ']
+  ];
+return true;
+} else
+return false;
 }
 
 function checkForWin() {
   // Your code here
+  //console.log("run 1");
+  horizontalWin();
+  verticalWin();
+  diagonalWin()
+  return true;
```

Comment:
The execution of this code is going to return true every time. Is there a win every time you put a mark on the board?